packages/psss-server/src/routes/reports/ConfirmCountryWeeklyReportRoute.ts

```diff
@@ -37,12 +56,93 @@ export class ConfirmCountryWeeklyReportRoute extends Route {
     const answers = mapUnconfirmedReportToConfirmedAnswers(report.results[0]);
 
     return this.meditrakConnection?.updateOrCreateSurveyResponse(
-      SURVEY_CODE,
+      CONFIRMED_WEEKLY_SURVEY_COUNTRY,
       countryCode,
       week,
       answers,
     );
   }
+
+  async createAlerts(countryCode: string, week: string) {
+    const report = await this.reportConnection?.fetchReport(
+      CONFIRMED_WEEKLY_REPORT_CODE,
+      [countryCode],
+      [week],
+    );
+
+    if (!report || report.results.length === 0) {
+      throw new RespondingError(
+        `Cannot create alerts: no confirmed weekly data found for ${countryCode} - ${week}`,
+        500,
+      );
+    }
+
+    const [result] = report.results;
+    const response: any = {
+      createdAlerts: [],
+    };
+    const startWeek = dateStringToPeriod(MIN_DATE, 'WEEK');
+    const activeAlertsData = await this.reportConnection?.fetchReport(
+      ACTIVE_ALERTS_REPORT_CODE,
+      [countryCode],
+      [startWeek, week],
+    );
+
+    if (!activeAlertsData) {
+      // should not be undefined even if there is no data
+      throw new RespondingError(
+        `Cannot create alerts: no active alerts data found for ${countryCode} - ${week}`,
+        500,
+      );
+    }
+
+    const { results: alerts } = activeAlertsData;
+    const alertsBySyndrome = groupBy(alerts, 'syndrome');
+
+    for (const syndromeCode of SYNDROME_CODES) {
+      const syndromeAlerts = alertsBySyndrome[syndromeCode];
+
+      // If there is no existing active alert for this syndrome,
+      // and the threshold is crossed for this syndrome, create a new one
+      if (!syndromeAlerts && result[`${syndromeCode} Threshold Crossed`] === true) {
+        const surveyResponseId = generateId();
+        await this.meditrakConnection?.createSurveyResponse(
+          ALERT_SURVEY,
+          countryCode,
+          week,
+          {
+            PSSS_Alert_Syndrome: syndromeCode,
+            PSSS_Alert_Archived: 'No',
+          },
+          surveyResponseId,
+        );
+
+        response.createdAlerts.push({
+          id: surveyResponseId,
+          title: syndromeCode,
+        });
+
+        continue;
+      }
+
+      const currentWeekSyndromeAlert =
+        syndromeAlerts && syndromeAlerts.find(a => a.period === week);
+
+      // If there is an existing alert triggered in the selected week,
+      // and now for the selected week, the threshold is no longer crossed (because of reconfirming changed data),
+      // archive the existing alert triggered in the selected week
+      if (currentWeekSyndromeAlert && result[`${syndromeCode} Threshold Crossed`] === false) {
+        const alertSurveyResponse = await this.meditrakConnection?.findSurveyResponseById(
+          currentWeekSyndromeAlert.id,
+        );
```

Comment:
```suggestion
```
I think this may be a bit of an un-needed fetch here. To get the `SurveyResponseObject` required by `updateSurveyResponse()` we only need `id`, `entityCode`, `dataTime` and `surveyCode`. All of these I believe we should have access to without requiring a fetch from meditrak-server correct?

We have a couple of options of how we could handle this:
a) Contruct a `SurveyResponseObject` inline here and pass it in
b) Refactor `updateSurveyResponse` to instead take `id`, `entityCode`, `dataTime` and `surveyCode`

I have a slight preference for b), how do you feel about it?

Reply:
I think it will be useful to have both? It would be good if TypeScript can do method overloading. But if picking 1 then I agree b is the better option.

Reply:
TS can actually do overloading 👍 great idea, both is good! https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

Reply:
After playing with overloading in TS, I generally feel it is a bit awkward (at least for my case) and it's quite different than how other programming languages do overloading. So I decided to have just different method names for both options. We need a different method that takes `id`, `entityCode`, `period` and `surveyCode` anyway because we need to do timezone stripping for the period, and for the method that takes surveyResponseObject, we don't need to because data_time timezone should already be stripped.

We can discuss this further in the office tomorrow

Reply:
Yeah that's all good, I also am not the biggest fan of function overloads in `TS`. Happy with your latest code 👍